gnomad_mitochondria/manuscript_analyses/generate_Fig1A_4_3BCD_S4ABC_S5.R

```diff
@@ -0,0 +1,688 @@
+grouped_counts$variant_group <- gsub("_true", "\n(haplogroup-defining)", grouped_counts$variant_group)
+grouped_counts$variant_group <- gsub("_false", "\n(not haplogroup-defining)", grouped_counts$variant_group)
+
+# Count number of variants split by variant (singleton, doubleton %, etc)
+grouped_by_variant <- variant_data %>% group_by(variant_af_group) %>% summarise(count=n()) %>% mutate(proportion = count/sum(count))
+
+
+# Create function to plot barplots for population allele frequencies
+plot_barplot <- function(df, x_variable, y_variable, fill_variable){
+  # Generate bar plot of df with the speicfied data for x and y axis
+  # Input 'df' is dataframe to be plotted
+  # x_variable is name of variable to use on x axis
+  # y_variable is name of variable to use on y axis
+  # fill_variable is name of variable to use for fill
+  
+  bar_plot <- ggplot(df, aes_string(x = x_variable, y = y_variable, fill = fill_variable)) + 
+    geom_bar(stat = "identity") +
+    theme_bw() +
+    theme(axis.text.x = element_text(size = 12, colour = "black", angle = 35, hjust = 1),
+          axis.text.y = element_text(size = 12, colour = "black"),
+          axis.title = element_text(colour = "black", size = 15, face = "bold"),
+          axis.title.x = element_blank(),
+          panel.grid.major = element_blank(), 
+          panel.grid.minor = element_blank(),
+          panel.background = element_blank(),
+          legend.key.size = unit(1.75, 'lines'),
+          legend.title = element_blank(),
+          plot.margin = unit(c(0.1, 0.1, 1, 0.1), "cm")) +
+    scale_fill_manual(values = c("chartreuse4", "chartreuse1", "cadetblue2", "dodgerblue4")) +
+    scale_x_discrete(limits = c("singleton", "doubleton-0.1%", "0.1-1%", "1-10%", ">10%"))
+  
+  return(bar_plot)
+}
+
+# Make barplot for proportion of unique variants falling into each frequency category
+pop_af_by_variant_plot <- plot_barplot(grouped_by_variant, "variant_af_group", "proportion", NULL) +
+  labs(x = "Percent of samples with variant", y = "Proportion of\nunique variants")
+pop_af_by_variant_plot
+
+# Now add in het/hom and whether or not a haplogroup-defining variant, and plot proportion of each frequency category that falls into these groups
+grouped_by_category <- grouped_counts %>% left_join(grouped_by_variant, by = "variant_af_group")
+grouped_by_category <- mutate(grouped_by_category, proportion = count.x/count.y)
+grouped_by_category$variant_group <- factor(grouped_by_category$variant_group,
+                                            levels = c(
+                                              "heteroplasmic\n(haplogroup-defining)",
+                                              "heteroplasmic\n(not haplogroup-defining)",
+                                              "homoplasmic\n(not haplogroup-defining)",
+                                              "homoplasmic\n(haplogroup-defining)"
+                                            )
+)
```

Comment:
Noting the grouping for manuscript figure slightly different (heteroplasmic collapsed)

Reply:
Yeah looks like this was tweaked downstream